### PR TITLE
Added upper limits of required dependencies

### PR DESCRIPTION
**Pull request by DanRCM**
#174
**Issue 1: 'ghc-options: -O2' is rarely needed:**
Time tests were conducted using `time cabal build`, yielding the following results:
Without using "-O2":
image

With the use of "-O2":
image

From these results, we can see that the use of "-O2" doesn't actually provide any time-related benefits.

**Issue 2: Missing Upper Bounds in Dependency Packages**
Upper bounds were added to the dependencies that required them: "-compat -containers -directory -hspec -shellwords -text". The `gen-bounds` tool was used, as suggested. Tests were conducted via `stack` to assess the compatibility of each of the packages with the upper bounds I set, based on the proposed suggestions using the `gen-bounds` tool."

### CODE REVIEW

**Comment by CristhianMotoche** [comment]
Thank you so much for your contribution @DanRCM LGTM!

File: dotenv.cabal [hunk 1]
```diff
@@ -66,9 +66,9 @@ executable dotenv
                        , base-compat >= 0.4
                        , dotenv
                        , optparse-applicative >= 0.11 && < 0.19
-                       , megaparsec
-                       , process
-                       , text
+                       , megaparsec >= 9.2.2 && <= 9.4.1
+                       , process >= 1.6.16 && < 1.7
+                       , text >= 1.2.5 && < 1.3
```
Comment (CristhianMotoche) [inline dotenv.cabal:71]: @DanRCM I  see we're also using this dependencies in the library. It's not necessary to define the boundaries for every dependency in library, executable, and test-suite. It's good practice to define them in one section it once and the boundaries will automatically be inherited by other sections.

File: dotenv.cabal [hunk 2]
```diff
@@ -119,10 +119,10 @@ test-suite dotenv-test
...
   build-depends:       base >= 4.9 && < 5.0
                      , dotenv
-                     , megaparsec
-                     , hspec
-                     , process
-                     , text
+                     , megaparsec >= 9.2.2 && <= 9.4.1
+                     , hspec >= 2.9.7 && <= 2.11.4
+                     , process >= 1.6.16 && < 1.7
+                     , text >= 1.2.5 && < 1.3
```
Comment (CristhianMotoche) [inline dotenv.cabal:125]: The [comment above](https://github.com/stackbuilders/dotenv-hs/pull/176/files#r1312448517) also applies here.

File: stack.yaml [hunk 3]
```diff
@@ -1 +1,3 @@
 resolver: lts-20.11
+
+
```
Comment (CristhianMotoche) [inline stack.yaml:3]: We can remove these extra lines.

**Comment by CristhianMotoche** [comment]
@DanRCM Let's specify the boundaries of the common dependencies only in the library section. The boundaries will be automatically inherited to the other sections.

File: dotenv.cabal [hunk 4]
```diff
@@ -91,13 +91,13 @@ library
...
                        , Configuration.Dotenv.Types
 
-  build-depends:         base >= 4.9 && < 5.0
-                       , directory
-                       , megaparsec >= 7.0.1 && < 10.0
-                       , containers
-                       , process >= 1.6.3.0 && < 1.7
-                       , shellwords >= 0.1.3.0
-                       , text
+  build-depends:         base >= 4.16.4 && < 4.17
```
Comment (CristhianMotoche) [inline dotenv.cabal:94]: @DanRCM Updating `base` version mean that you're limiting the version of GHC that is compatible with this library. You can check the relation between GHC and base [here](https://wiki.haskell.org/Base_package). `dotenv` is tested against GHC 8.10 (base >= 4.14). Let's not change that boundary.

**Review by CristhianMotoche — COMMENTED** [review]
@DanRCM It looks good. Let's not update the boundaries of `base`, though.

**Comment by DanRCM** [comment]
@CristhianMotoche Hi, thanks for your recommendation, the changes have been made.

**Comment by CristhianMotoche** [comment]
Hello @DanRCM Thanks for addressing the suggestions. It seems something in the recent changes has caused issues when building the package. I think you need to lower the boundaries for `text`.

**Comment by CristhianMotoche** [comment]
Hello @DanRCM Do you need any help on this PR?

**Comment by DanRCM** [comment]
> Hello @DanRCM Do you need any help on this PR?

Hello, excuse me, I have been really busy, but I'll continue working in your project, thank you for your help, I'll do the changues inmediately

**Comment by DanRCM** [comment]
@CristhianMotoche  The changes have been made, I have been taking your advices to lower the `text` boundaries.

**Comment by DanRCM** [comment]
@CristhianMotoche
Hello, could you tell me what you suspect about what could be giving me problems with the RP? I need help

**Comment by CristhianMotoche** [comment]
Hello @DanRCM It seems there are some issues with the changed boundaries. Let me double check your PR locally and I'll keep you posted.

**Comment by DanRCM** [comment]
> Hello @DanRCM It seems there are some issues with the changed boundaries. Let me double check your PR locally and I'll keep you posted.

ok, thanks for keeping me in mind, it's a pleasure

**Review by CristhianMotoche — APPROVED** [review]

**Comment by CristhianMotoche** [comment]
@all-contributors please add @DanRCM  for code

**Pull request merged**